src/ImageSharp/Formats/Jpeg/Components/FastFloatingPointDCT.Intrinsic.cs

```diff
@@ -0,0 +1,243 @@
+
+            Vector128<float> z1 = Sse.Multiply(Sse.Add(tmp12, tmp13), mm128_F_0_7071);
+            Unsafe.Add(ref blockRef, 4) = Sse.Add(tmp13, z1);
+            Unsafe.Add(ref blockRef, 12) = Sse.Subtract(tmp13, z1);
+
+            // Odd part
+            tmp10 = Sse.Add(tmp4, tmp5);
+            tmp11 = Sse.Add(tmp5, tmp6);
+            tmp12 = Sse.Add(tmp6, tmp7);
+
+            Vector128<float> z5 = Sse.Multiply(Sse.Subtract(tmp10, tmp12), mm128_F_0_3826);
+            Vector128<float> z2 = Sse.Add(Sse.Multiply(mm128_F_0_5411, tmp10), z5);
+            Vector128<float> z4 = Sse.Add(Sse.Multiply(mm128_F_1_3065, tmp12), z5);
+            Vector128<float> z3 = Sse.Multiply(tmp11, mm128_F_0_7071);
+
+            Vector128<float> z11 = Sse.Add(tmp7, z3);
+            Vector128<float> z13 = Sse.Subtract(tmp7, z3);
+
+            Unsafe.Add(ref blockRef, 10) = Sse.Add(z13, z2);
+            Unsafe.Add(ref blockRef, 6) = Sse.Subtract(z13, z2);
+            Unsafe.Add(ref blockRef, 2) = Sse.Add(z11, z4);
+            Unsafe.Add(ref blockRef, 14) = Sse.Subtract(z11, z4);
+        }
+
+        /// <summary>
+        /// Apply 1D floating point FDCT inplace using AVX operations on 8x8 matrix.
+        /// </summary>
+        /// <remarks>
+        /// Requires Avx support.
+        /// </remarks>
+        /// <param name="block">Input matrix.</param>
+        public static void FDCT8x8_avx(ref Block8x8F block)
+        {
+            DebugGuard.IsTrue(Avx.IsSupported, "Avx support is required to execute this operation.");
+
+            Vector256<float> tmp0 = Avx.Add(block.V0, block.V7);
+            Vector256<float> tmp7 = Avx.Subtract(block.V0, block.V7);
+            Vector256<float> tmp1 = Avx.Add(block.V1, block.V6);
+            Vector256<float> tmp6 = Avx.Subtract(block.V1, block.V6);
+            Vector256<float> tmp2 = Avx.Add(block.V2, block.V5);
+            Vector256<float> tmp5 = Avx.Subtract(block.V2, block.V5);
+            Vector256<float> tmp3 = Avx.Add(block.V3, block.V4);
+            Vector256<float> tmp4 = Avx.Subtract(block.V3, block.V4);
+
+            // Even part
+            Vector256<float> tmp10 = Avx.Add(tmp0, tmp3);
+            Vector256<float> tmp13 = Avx.Subtract(tmp0, tmp3);
+            Vector256<float> tmp11 = Avx.Add(tmp1, tmp2);
+            Vector256<float> tmp12 = Avx.Subtract(tmp1, tmp2);
+
+            block.V0 = Avx.Add(tmp10, tmp11);
+            block.V4 = Avx.Subtract(tmp10, tmp11);
+
+            Vector256<float> z1 = Avx.Multiply(Avx.Add(tmp12, tmp13), mm256_F_0_7071);
+            block.V2 = Avx.Add(tmp13, z1);
+            block.V6 = Avx.Subtract(tmp13, z1);
+
+            // Odd part
+            tmp10 = Avx.Add(tmp4, tmp5);
+            tmp11 = Avx.Add(tmp5, tmp6);
+            tmp12 = Avx.Add(tmp6, tmp7);
+
+            Vector256<float> z5 = Avx.Multiply(Avx.Subtract(tmp10, tmp12), mm256_F_0_3826);
+            Vector256<float> z2 = Avx.Add(Avx.Multiply(mm256_F_0_5411, tmp10), z5);
```

Comment:
Not to myself: replace all such mul/add with fma.

Reply:
We have a helper for that in `Numerics`.